daemon/containerd/image_exporter.go

```diff
@@ -85,7 +83,17 @@ func (i *ImageService) ExportImage(ctx context.Context, names []string, outStrea
 		return leaseContent(ctx, i.content, leasesManager, lease, target)
 	}
 
-	exportImage := func(ctx context.Context, target ocispec.Descriptor, ref reference.Named) error {
+	exportImage := func(ctx context.Context, img containerdimages.Image, ref reference.Named) error {
+		target := img.Target
+
+		if platform != nil {
+			newTarget, err := i.getPushDescriptor(ctx, img, platform)
+			if err != nil {
+				return errors.Wrap(err, "no suitable export target found")
```

Comment:
- Would it make sense to include `platform` here somehow (to make clear that there's no image found for the given platform)? Or does `getPushdescriptor` already return this?
- Do we need an `errdefs`  type here?

Reply:
Ah, good catch. Added platform in message here.
For `errdefs` - `getPushDescriptor` is expected to return an errdefs already, fixed a case where it didn't.